docs/r3/beyondcandc.md

```diff
@@ -32,16 +33,22 @@ A well-known centralised mechanism for reliable synchronisation of state changes
 * **Centralised state and trust**: the transaction manager keeps track of the state of the two-phase commit, and is trusted by both sides to advance that state in accordance with the protocol.
 * **Trusted agents or centralised verification**: either each participant in the cross-network transaction must trust that that the other will accurately report the success or failure of commands issued by the transaction manager, or the transaction manager must be able to obtain and verify proofs from both sides that the actions it has requested have been carried out.
 
-A decentralised solution might begin by removing the centralised intermediary and making one network responsible for managing the state of the swap protocol, for example through a contract on an EVM blockchain.
+A decentralised solution might begin by removing the centralised intermediary and making one network responsible for managing the state of the swap protocol, for example through a contract on an EVM blockchain. The contract itself, supported by the EVM network’s validation and consensus rules, would provide guarantees that the state of the protocol evolves correctly. The challenge in this case would be enabling that contract to collect and verify "votes" from each side.
+
+## The Corda-EVM cross-chain swap pattern
 
-The contract itself, supported by the EVM network’s validation and consensus rules, could provide guarantees that the state of the protocol evolves correctly. The challenge in this case would be enabling that contract to verify that its commands have been completed properly by both sides.
+In our proposed Corda-EVM cross-chain swap pattern there is no separate EVM contract co-ordinating the swap. The working state of the protocol is represented by hold states created, and then released, on each network’s ledger. However, the basic shape of the pattern remains the same:
 
-In the solution we discuss here there is no separate EVM contract co-ordinating the swap. The working state of the protocol is represented by locks created, and then released, on each network’s ledger.
+1. The Corda network and EVM network both place assets in "hold" states: a "committed" state on the EVM side, and a "locked" state on the Corda side.
+2. The "committed" state on the EVM side is set up so that the receiver can transfer the asset to themselves on presentation of proof that the "locked" state was created on the Corda side. When the EVM transfer transaction takes place, this automatically creates a publicly broadcast proof of action that the receiver on the Corda side can use to obtain their asset from the Corda lock.
+3. If the receiver on the EVM side does not perform the transfer, either the sender or receiver can perform a revert sending the asset back to its original owner; this likewise creates a publicly broadcast proof of action that the sender on the Corda side can use to retrieve their asset from the Corda lock.
```

Comment:
true - but the denial of state is precisely the thing we worry about, no?   In the full-broadcast (EVM) model, the act of getting your 'thing' confirmed is indivisible from everybody else eventually being guaranteed to learn about it.  The same is not true of Corda.  eg the party on Corda might go offline for an indefinite period after getting the notarisation but before completing the obligations of the finality flow.

But maybe C5 works differently?

Reply:
The denial of state problem remains essentially unsolved in this pattern and in any extant implementation of Corda. We might in the future have a Notary that provides some extra functionality that mitigates it.

If Alice@USD notarises the draft tx and uses the notary's signature to claim the asset on the EVM side, but does not share the notary's signature with Bob, then Bob doesn't hold the Tx creating the lock state and can't use the proof of action to unlock it and get the asset.

Reply:
exactly - that's my point.  And hence why we set things up very carefully so that the one time when 'i need to know when the other side has done something even though it's not in their interest to tell me' occurs, it happens on the EVM side.  So we take advantage of the EVM's relatively lower privacy promise and, at the same time, mitigate the flipside of Corda's stronger promise.